modules/http/http.c

```diff
@@ -261,6 +263,66 @@ http_dd_set_ssl_version(LogDriver *d, const gchar *value)
   return TRUE;
 }
 
+static inline gboolean
+_http_dd_curl_compression_string_match(const gchar *string, gint curl_compression_index)
+{
+  return (strcmp(string, curl_compression_types[curl_compression_index]) == 0);
+}
+
+gboolean
+_http_dd_check_curl_compression(const gchar *type)
+{
+  if(_http_dd_curl_compression_string_match(type, CURL_COMPRESSION_UNCOMPRESSED)) return TRUE;
+#if defined(CURL_COMPRESSION_IMPLEMENTED_GZIP) && defined(ZLIB_AVAILABLE)
+  if(_http_dd_curl_compression_string_match(type, CURL_COMPRESSION_GZIP)) return TRUE;
+#endif
+#if defined(CURL_COMPRESSION_IMPLEMENTED_DEFLATE) && defined(ZLIB_AVAILABLE)
+  if(_http_dd_curl_compression_string_match(type, CURL_COMPRESSION_DEFLATE)) return TRUE;
+#endif
+#if defined(CURL_COMPRESSION_IMPLEMENTED_BROTLI) && defined(BROTLI_AVAILABLE)
+  if(_http_dd_curl_compression_string_match(type, CURL_COMPRESSION_BROTLI)) return TRUE;
+#endif
+#if defined(CURL_COMPRESSION_IMPLEMENTED_ZSTD) && defined(ZSTD_AVAILABLE)
+  if(_http_dd_curl_compression_string_match(type, CURL_COMPRESSION_ZSTD)) return TRUE;
+#endif
+  return FALSE;
+}
```

Comment:
Seeing that `http-curl-compression.{c,h}` gives both the enum and the string representation, I think the responsibility of check/match/convert between them should be there.

A similar pattern can be seen with `_LogMessageValueType`, `log_msg_value_type_to_str()` and `log_msg_value_type_from_str()` in `lib/logmsg/logmsg.{c,h}`.

Reply:
There are still responsibilites and inner logic leaking from the `compression` module, like exposing and using `curl_compression_types` and calling check manually on the user side.

With your current implementation the users of the compression module need to know inner knowledge about the enum value matching to the position in the `curl_compression_types` array. They can also forget to check if the compression is supported with the user's build, this leaves places for error. I think all of these responsibilities should be in the compression module.

I think everything can be done with only two functions in the compression module:
```
gboolean compression_type_from_str(const gchar *in_string, CurlCompressionType *out_compression_type);
const gchar *compression_type_to_str(CurlCompressionType compression_type);
```

`compression_type_from_str()` can return `FALSE`, if the input string is invalid or the corresponding compression type is not supported with the user's build.
`compression_type_to_str()` is a simple C-style mapping from `enum` to `const char*`.

For an example implementation, see: https://github.com/syslog-ng/syslog-ng/blob/master/lib/logmsg/logmsg.c#L233